cls/SourceControl/Git/DiscardState.cls

```diff
@@ -0,0 +1,138 @@
+Class SourceControl.Git.DiscardState Extends (%Persistent, %JSON.Adaptor)
```

Comment:
An issue I'm just noticing here: this won't play well with package manager-based editing of multiple packages with distinct repo roots in the same namespace. I'd be happy to say we can't do that just yet and file another issue to pick it up in the future (not necessarily for 2.6.0).